lib/rules/object-shorthand.js

```diff
@@ -231,59 +360,31 @@ module.exports = {
                 }
 
                 // only computed methods can fail the following checks
-                if (node.computed && node.value.type !== "FunctionExpression") {
+                if (node.computed && node.value.type !== "FunctionExpression" && node.value.type !== "ArrowFunctionExpression") {
                     return;
                 }
 
                 //--------------------------------------------------------------
                 // Checks for property/method shorthand.
                 if (isConciseProperty) {
+                    if (node.method && (APPLY_NEVER || AVOID_QUOTES && isStringLiteral(node.key))) {
 
-                    // if we're "never" and concise we should warn now
-                    if (APPLY_NEVER) {
-                        const type = node.method ? "method" : "property";
-
+                        // { x() {} } should be written as { x: function() {} }
                         context.report({
                             node,
-                            message: "Expected longform {{type}} syntax.",
-                            data: {
-                                type
-                            },
-                            fix(fixer) {
-                                if (node.method) {
-                                    if (node.value.generator) {
-                                        return fixer.replaceTextRange([node.range[0], node.key.range[1]], `${node.key.name}: function*`);
-                                    }
-
-                                    return fixer.insertTextAfter(node.key, ": function");
-                                }
-
-                                return fixer.insertTextAfter(node.key, `: ${node.key.name}`);
-                            }
+                            message: `Expected longform method syntax${APPLY_NEVER ? "" : " for string literal keys"}.`,
```

Comment:
I guess that makes sense - it seems a confusing term tho, since "shorthand" is a shorter form of "normal", and "long form" would be a longer form of "normal".